### ISSUE TITLE
Save the colors!!!

### ISSUE THREAD

**Issue by voronianski** [DennisSnijder/MakeGithubGreatAgain#49]
I think you need to save us from new github colors! Especially links!

**Comment by deepsweet** [comment]
https://mobile.twitter.com/djspiewak/status/836985679854845952

**Comment by ricordisamoa** [comment]
https://github.com/ricordisamoa/MakeGithubPaleAgain

**Comment by sedders123** [comment]
Found the old link colour from a stale tab #4078c0

**Comment by sedders123** [comment]
And the css for the private label
```
.label-private {
    font-weight: normal;
    color: #4c4a42;
    background-color: #ffefc6;
}
```

**Comment by ricordisamoa** [comment]
I am sorry that my comment has been dumped like it were spam. I was just linking to a small repository where I collected several of the old colors like the ones mentioned above, therefore I presumed that could have solved this issue (even if not in this repository). I apologize.

**Comment by maximilianhurl** [comment]
The new green buttons are particularly upsetting on my monitor. Here's the old CSS for reference:

    /* Green primary button */
    .btn-primary {
        color: #fff;
        text-shadow: 0 -1px 0 rgba(0,0,0,0.15);
        background-color: #6cc644;
        background-image: -webkit-linear-gradient(#91dd70, #55ae2e);
        background-image: linear-gradient(#91dd70, #55ae2e);
        border: 1px solid #5aad35;
    }

**Comment by rightfold** [comment]
Because the current software only applies to the header color, there should be a setting so that you can select which changes to apply. I like the old header with the new colors, for example.

**Comment by thieman** [comment]
@rightfold this half-effort appeasing attitude is completely against the stated goal of the project to make github GREAT again, just like it was in the 1950s

**Comment by svnpenn** [comment]
Here is a reverse diff, with the old good colors in green, and the new bad
colors in red:

http://github.com/svnpenn/dotfiles/blob/f8fe891/stylish/save-the-colors.diff

**Comment by DennisSnijder** [comment]
We need to make the colors great again!
Feel free to shoot in a PR 👍

**Comment by miyaoka** [comment]
http://web.archive.org/web/*/github.com

![2008](https://cloud.githubusercontent.com/assets/1443118/23554202/6df669c8-0067-11e7-9950-5239ff3165ef.png)
![2013](https://cloud.githubusercontent.com/assets/1443118/23554206/727c7e60-0067-11e7-9133-6ed284c6143e.png)
![2014](https://cloud.githubusercontent.com/assets/1443118/23554347/092be044-0068-11e7-8f0f-26908c3720a2.png)

We should support github GREAT 2008 again, GREAT 2009 again, GREAT 2010 again...

**Comment by svnpenn** [comment]
https://github.com/DennisSnijder/MakeGithubGreatAgain/pull/50

**Comment by joeytwiddle** [comment]
Just found this extension: [github-old-colors](https://chrome.google.com/webstore/detail/github-old-colors/njodkcccjccenllcjfpomnokbkigpcii)

So far it only changes links back to the old blue, and removes some shadows and background images from buttons. It doesn't restore the header bar at all!

It uses a whitelist for links, which might be less problematic than the blacklist in my PR. This is an option worth considering...

**Comment by svnpenn** [comment]
@joeytwiddle i dont get why your pull has both save-the-colors.css and links.css

**Comment by joeytwiddle** [comment]
@svnpenn In my Chrome, our custom styles are injected before the page styles load. As a result, the page's `a` rule overrides the `a` rule in save-the-colors.css

![page_styles_mask_injected](https://cloud.githubusercontent.com/assets/911799/23894988/2aaf0820-08df-11e7-8403-9d31b6ddfd17.png)

That's why I took tombola's script for styling the links (and for the toggling behaviour).

For sure, the two scripts could be combined into one.

The larger concern for me now is how to override their CSS without accidentally styling elements we don't want.

I'm beginning to think a more workable approach would be to inject styles _after_ the page's styles, and abandon toggling between new/old colors. (Either that, or doing a dynamic search-replace on the CSSDOM after the page's styles load.)

**Comment by joeytwiddle** [comment]
On Firefox, I'm giving these a try:

- https://userstyles.org/styles/138766/github-return-light-themed-header
- https://userstyles.org/styles/139586/github-softer-blue-color

They don't restore the old greens and purples. But they do restore the old white header and many of the blue links.

**Comment by DennisSnijder** [comment]
Fixed by #51
